network/src/validator_network/network_builder.rs

```diff
@@ -414,15 +414,26 @@ impl NetworkBuilder {
             .take()
             .expect("Authentication Mode not set");
 
-        let (key, maybe_trusted_peers) = match authentication_mode {
-            AuthenticationMode::ServerOnly(key) => (key, None),
-            AuthenticationMode::Mutual(key) => (key, Some(self.trusted_peers.clone())),
+        let (key, maybe_trusted_peers, peer_id) = match authentication_mode {
+            // validator-operated full node
+            AuthenticationMode::ServerOnly(key) if self.peer_id == PeerId::default() => {
```

Comment:
it looks like `peer_id` should be `Option<PeerId>` in the config then? these comments also don't match up with my current mental model for the different node types. hopefully we can clarify this in tomorrow's call.

anyway, I'm fine putting this off. I'm partly blocked on this so let's just land it now?